anyschema/_anyschema.py

```diff
@@ -134,6 +134,11 @@ def __init__(
                     for name, input_type, metadata in into_ordered_dict_adapter(spec)
                 }
             )
+        elif is_dataclass(spec):
```

Comment:
This would include also [pydantic dataclasses](https://docs.pydantic.dev/latest/concepts/dataclasses/), can you add a test for them as well?